src/Microsoft.DotNet.Build.CloudTestTasks/AzureHelper.cs

```diff
@@ -0,0 +1,373 @@
+        }
+
+        /// <summary>
+        /// The get canonicalized headers.
+        /// </summary>
+        /// <param name="request">
+        /// The request.
+        /// </param>
+        /// <returns>
+        /// The <see cref="string"/>.
+        /// </returns>
+        public static string GetCanonicalizedHeaders(HttpWebRequest request)
+        {
+            var sb = new StringBuilder();
+            var headerNameList = (from headerName in request.Headers.AllKeys
+                                  where headerName.ToLowerInvariant().StartsWith("x-ms-", StringComparison.Ordinal)
+                                  select headerName.ToLowerInvariant()).ToList();
+            headerNameList.Sort();
+            foreach (var headerName in headerNameList)
+            {
+                var builder = new StringBuilder(headerName);
+                var separator = ":";
+                foreach (var headerValue in GetHeaderValues(request.Headers, headerName))
+                {
+                    var trimmedValue = headerValue.Replace("\r\n", string.Empty);
+                    builder.Append(separator);
+                    builder.Append(trimmedValue);
+                    separator = ",";
+                }
+
+                sb.Append(builder);
+                sb.Append("\n");
+            }
+
+            return sb.ToString();
+        }
+
+        /// <summary>
+        /// The get canonicalized resource.
+        /// </summary>
+        /// <param name="address">
+        /// The address.
+        /// </param>
+        /// <param name="accountName">
+        /// The account name.
+        /// </param>
+        /// <returns>
+        /// The <see cref="string"/>.
+        /// </returns>
+        public static string GetCanonicalizedResource(Uri address, string accountName)
+        {
+            var str = new StringBuilder();
+            var builder = new StringBuilder("/");
+            builder.Append(accountName);
+            builder.Append(address.AbsolutePath);
+            str.Append(builder);
+            var queryKeyValues = ExtractQueryKeyValues(address);
+            var dictionary = GetCommaSeparatedList(queryKeyValues);
+
+            foreach (var pair in dictionary.OrderBy(p => p.Key))
+            {
+                var stringBuilder = new StringBuilder(string.Empty);
+                stringBuilder.Append(pair.Key + ":");
+                var commaList = string.Join(",", pair.Value);
+                stringBuilder.Append(commaList);
+                str.Append("\n");
+                str.Append(stringBuilder);
+            }
+
+            return str.ToString();
+        }
+
+        /// <summary>
+        /// The get header values.
+        /// </summary>
+        /// <param name="headers">
+        /// The headers.
+        /// </param>
+        /// <param name="headerName">
+        /// The header name.
+        /// </param>
+        /// <returns>
+        /// The <see>
+        ///         <cref>List</cref>
```

Comment:
I think this was supposed to be `The <see cref="List" />`?  It's gone pretty wonky...